pkg/multishare_reconciler/reconciler.go

```diff
@@ -0,0 +1,873 @@
+						continue
+					}
+					klog.Infof("share %q is now assigned to instance %q", shareInfo.Name, instanceURI)
+					instanceInfos[instanceURI] = instanceInfo
+					break
+				}
+			}
+			if err != nil {
+				continue
+			}
+
+			if instanceURI == "" {
+				klog.Infof("couldn't find instance to fit share %q, generating new instance", shareInfo.Name)
+
+				instanceURI, _ = file.GenerateMultishareInstanceURI(&file.MultishareInstance{
+					Project:  recon.cloud.Project,
+					Location: shareInfo.Spec.Region,
+					Name:     util.NewMultishareInstancePrefix + string(uuid.NewUUID()),
+				})
+
+				instanceInfo, err := recon.requestInstance(instanceURI, shareInfo.Spec.InstancePoolTag)
+				if err != nil {
+					klog.Errorf("failed to create new instanceInfo %q: %v", instanceURI, err)
+					continue
+				}
+				instanceInfo, err = recon.assignInstanceInfo(instanceInfo, shareInfo.Name)
+				if err != nil {
+					klog.Errorf("failed to add share %q to instanceInfo %q: %v", shareInfo.Name, instanceInfo.Name, err)
+					continue
+				}
+				klog.Infof("share %q is now assigned to instance %q", shareInfo.Name, instanceURI)
+				instanceInfos[instanceURI] = instanceInfo
+			}
+
+			shareInfo, err := recon.assignShareInfo(shareInfo, instanceURI)
+			if err != nil {
+				klog.Errorf("cannot update the instanceHandle of shareInfo %q: %v", shareInfo.Name, err)
+				continue
+			}
+			shareInfos[shareInfo.Name] = shareInfo
+		}
+	}
+
+	for instanceURI, instanceInfo := range instanceInfos {
+		if instanceInfo.DeletionTimestamp != nil {
+			continue
+		}
+
+		instanceInfoClone := instanceInfo.DeepCopy()
+		var updated bool
+		if instanceEmpty(instanceInfo) {
+			instanceInfoClone, updated = maybeAddCleanupFinalizer(instanceInfoClone)
+			if updated {
+				klog.Infof("instanceInfo %q needs to be deleted, trying to add finalizer", instanceInfo.Name)
+				instanceInfoClone, err := recon.updateInstanceInfo(context.TODO(), instanceInfoClone)
+				if err != nil {
+					klog.Errorf("failed to update instanceInfo %q: %v", instanceInfo.Name, err)
+					continue
+				}
+				err = recon.deleteInstanceInfo(context.TODO(), instanceInfoClone)
+				if err != nil {
+					klog.Errorf("failed to add deletionTimestamp to instanceInfo %q: %v", instanceInfo.Name, err)
+					continue
+				}
+			}
+		} else {
+			instanceInfoClone, updated = recon.instanceInfoNewCapacity(instanceInfoClone)
+			if updated {
+				klog.Infof("instanceInfo %q has updated capacity, trying to update object", instanceInfo.Name)
+				instanceInfoClone, err := recon.updateInstanceInfo(context.TODO(), instanceInfoClone)
+				if err != nil {
+					klog.Errorf("failed to update instanceInfo %q: %v", instanceInfo.Name, err)
+					continue
+				}
+				instanceInfos[instanceURI] = instanceInfoClone
+			}
+		}
+	}
+}
+
+func (recon *multishareReconciler) requestInstance(instanceURI string, scTag string) (*v1alpha1.InstanceInfo, error) {
```

Comment:
comment what the function does

Reply:
changed the name to `generateInstanceInfo()`. I think since the `createInstanceInfo()` is always called after the instanceInfo is generated, there's no need to split

Reply:
Ack